source/module_base/para_json.cpp

```diff
@@ -0,0 +1,986 @@
+    // rapidjson::Value gamma_only;
+    // rapidjson::Value ks_solver;
+    // rapidjson::Value ntype;
+    // rapidjson::Value nspin;
+    // rapidjson::Value ecutwfc;
+    // rapidjson::Value scf_thr;
+    // rapidjson::Value scf_nmax;
+
+    // @param init -- symmetry
+    // rapidjson::Value symmetry(rapidjson::kObjectType);
+    // rapidjson::Value BRAVAIS_TYPE;
+    // rapidjson::Value BRAVAIS_LATTICE_NAME;
+    // rapidjson::Value IBRAV;
+    // rapidjson::Value LATTICE_CONSTANT_A;
+    // rapidjson::Value right_hand_lattice;
+
+    // @param init -- Kpoints
+    rapidjson::Value kpoints(rapidjson::kObjectType);
+    rapidjson::Value nkstot;
+    rapidjson::Value nkstot_ibz;
+    rapidjson::Value coordinates(rapidjson::kArrayType);
+    rapidjson::Value weight(rapidjson::kArrayType);
+
+    // @param init -- grid
+    rapidjson::Value grid(rapidjson::kObjectType);
+    rapidjson::Value energy_cutoff_for_wavefunc;
+    rapidjson::Value fft_grid_for_wave_functions(rapidjson::kArrayType);
+    rapidjson::Value number_of_plane_waves;
+    rapidjson::Value number_of_sticks;
+
+    // @param init -- Smearing
+    // rapidjson::Value smearing_method;
+    // rapidjson::Value smearing_sigma;
+
+    // @param init -- mixing
+    rapidjson::Value mixing;
+
+
+    // @param output
+    rapidjson::Value output(rapidjson::kArrayType);
+
+
+
+    // @param final_stru
+    rapidjson::Value final_stru(rapidjson::kObjectType);
+    rapidjson::Value cell;
+    rapidjson::Value coordinate;
+
+
+
+
+    /**
+     *  The functions below initialize the json output parameter 
+     *  tree to connect the nodes of the module
+    */
+
+    /**
+     * @brief   add Top stage：parameter in Abacus:
+     */
+    void Init_json_abacus()
+    {
+
+
+        // add First stage：parameter in abcus:
+
+        abacus.AddMember("general_info", general_info, doc.GetAllocator());
+
+        abacus.AddMember("readin_info", readin_info, doc.GetAllocator());
+        
+        abacus.AddMember("init", init, doc.GetAllocator());
+
+        abacus.AddMember("output", output, doc.GetAllocator());
+
+        abacus.AddMember("final_stru", final_stru, doc.GetAllocator());
+
+        doc.SetObject();
+        // abacus.SetObject();
+        doc.AddMember("ABACUS", abacus, doc.GetAllocator());
+        /**
+         * .
+         * .
+         * .
+         * .
+         * .
+         * .
+         * .
+         * */
+    }
+    /**
+     * @brief   add Second stage：parameter in Abacus - general_info:
+     */
+    void Init_json_abacus_generalInfo(){
+        general_info.AddMember("version", version, doc.GetAllocator());
+
+        general_info.AddMember("commit", commit, doc.GetAllocator());      
+
+        general_info.AddMember("begin_time", begin_time, doc.GetAllocator());      
+
+        general_info.AddMember("begin_date", begin_date, doc.GetAllocator());     
+
+        general_info.AddMember("device_g", device_g, doc.GetAllocator());                
```

Comment:
why "device_g"? "device"?

Reply:
Since the device variable appears in both readin_info and general_info, a suffix is added to the general_info device for the sake of distinction

Reply:
A suffix in code is ok, but we need the key in json file is "device"

Reply:
ok,I ignore that